File: summer/WebContent/js/system/windows/MouseOverProperty.js
/**
 * MouseOverProperty
 */

define(["dojo/_base/declare", "system/Type", "windows/ReverseInheritProperty", "windows/CoreFlags"], 
		function(declare, Type, ReverseInheritProperty, CoreFlags){
	var UIElement =null;
	function EnsureUIElement(){
		if(UIElement == null){
			UIElement = using("windows/UIElement");
		}
		return UIElement;
	}
	
	var MouseOverProperty = declare("MouseOverProperty", ReverseInheritProperty,{
   	 	"-chains-": {
	      constructor: "manual"
	    },
		constructor:function(){
			ReverseInheritProperty.prototype.constructor.call(this, 
					EnsureUIElement().IsMouseOverPropertyKey, 
					CoreFlags.IsMouseOverCache,
					CoreFlags.IsMouseOverChanged);
		},
		
//		 internal override void 
		FireNotifications:function(element, /*bool*/ oldValue)
        { 
            // Before we fire the mouse event we need to figure if the notification is still relevant.
            // This is because it is possible that the mouse state has changed during the previous 
            // property engine callout. Example: Consider a MessageBox being displayed during the 
            // IsMouseOver OnPropertyChanged override.
 
            var shouldFireNotification = false;
            shouldFireNotification = (!oldValue && element.IsMouseOver) || (oldValue && !element.IsMouseOver); 

//            if (shouldFireNotification) 
//            { 
//                var mouseEventArgs = new MouseEventArgs(Mouse.PrimaryDevice, Environment.TickCount, Mouse.PrimaryDevice.StylusDevice);
//                mouseEventArgs.RoutedEvent = oldValue ? Mouse.MouseLeaveEvent : Mouse.MouseEnterEvent; 
//
//                element.RaiseEvent(mouseEventArgs); 
//            }
        } 
	});
	
	MouseOverProperty.Type = new Type("MouseOverProperty", MouseOverProperty, [ReverseInheritProperty.Type]);
	return MouseOverProperty;
});
 
        /////////////////////////////////////////////////////////////////////

       

File: org.summer.view.widget/src/org/summer/view/widget/MouseOverProperty.java
package org.summer.view.widget;

import org.summer.view.widget.input.Mouse;
import org.summer.view.widget.input.MouseEventArgs;
 ///////////////////////////////////////////////////////////////////////// 
 
    /*internal*/ public class MouseOverProperty extends ReverseInheritProperty
    { 
        /////////////////////////////////////////////////////////////////////

        /*internal*/ public MouseOverProperty()  
        { 
        	super(UIElement.IsMouseOverPropertyKey, 
                    CoreFlags.IsMouseOverCache,
                    CoreFlags.IsMouseOverChanged);
        }
 
        /////////////////////////////////////////////////////////////////////

        /*internal*/ public /*override*/ void FireNotifications(UIElement uie, ContentElement ce, UIElement3D uie3D, boolean oldValue)
        { 
            // Before we fire the mouse event we need to figure if the notification is still relevant.
            // This is because it is possible that the mouse state has changed during the previous 
            // property engine callout. Example: Consider a MessageBox being displayed during the 
            // IsMouseOver OnPropertyChanged override.
 
            boolean shouldFireNotification = false;
            if (uie != null)
            {
                shouldFireNotification = (!oldValue && uie.IsMouseOver) || (oldValue && !uie.IsMouseOver); 
            }
            else if (ce != null) 
            { 
                shouldFireNotification = (!oldValue && ce.IsMouseOver) || (oldValue && !ce.IsMouseOver);
            } 
            else if (uie3D != null)
            {
                shouldFireNotification = (!oldValue && uie3D.IsMouseOver) || (oldValue && !uie3D.IsMouseOver);
            } 

            if (shouldFireNotification) 
            { 
                MouseEventArgs mouseEventArgs = new MouseEventArgs(Mouse.PrimaryDevice, Environment.TickCount, Mouse.PrimaryDevice.StylusDevice);
                mouseEventArgs.RoutedEvent = oldValue ? Mouse.MouseLeaveEvent : Mouse.MouseEnterEvent; 

                if (uie != null)
                {
                    uie.RaiseEvent(mouseEventArgs); 
                }
                else if (ce != null) 
                { 
                    ce.RaiseEvent(mouseEventArgs);
                } 
                else if (uie3D != null)
                {
                    uie3D.RaiseEvent(mouseEventArgs);
                } 
            }
        } 
    } 